src/components/OrgFile/components/TaskListModal/index.js

```diff
@@ -0,0 +1,162 @@
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import './stylesheet.css';
+
+import TaskListView from './components/TaskListView';
+import Drawer from '../../../UI/Drawer';
+
+import { isMatch, computeCompletionsForDatalist } from '../../../../lib/headline_filter';
+
+import {
+  extractAllOrgTags,
+  extractAllOrgProperties,
+  extractAllTodoKeywords,
+} from '../../../../lib/org_utils';
+
+import parser from '../../../../lib/headline_filter_parser';
+
+import * as orgActions from '../../../../actions/org';
+
+import _ from 'lodash';
+import format from 'date-fns/format';
+
+class TaskListModal extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    _.bindAll(this, [
+      'handleHeaderClick',
+      'handleToggleDateDisplayType',
+      'handleFilterChange',
+      'handleSearchAllCheckboxChange',
+    ]);
+
+    this.state = {
+      selectedDate: new Date(),
+      dateDisplayType: 'absolute',
+      filterString: '',
+      filterExpr: [],
+      searchAllHeaders: false,
+    };
+  }
+
+  handleHeaderClick(headerId) {
+    this.props.onClose();
+    this.props.org.selectHeaderAndOpenParents(headerId);
+  }
+
+  handleToggleDateDisplayType() {
+    const { dateDisplayType } = this.state;
+
+    this.setState({
+      dateDisplayType: dateDisplayType === 'absolute' ? 'relative' : 'absolute',
+    });
+  }
+
+  handleSearchAllCheckboxChange(event) {
+    const searchAllHeaders = event.target.value;
+    this.setState({ searchAllHeaders });
+  }
+
+  handleFilterChange(event) {
+    const filterString = event.target.value;
+    const curserPosition = event.target.selectionStart;
+    this.setState({ filterString, curserPosition });
+    try {
+      const filterExpr = parser.parse(this.state.filterString);
+      // state change triggers rendering? -> infinite loop
+      this.setState({ filterExpr });
+    } catch (e) {
+      //console.log(e);
+      // TODO: highlight the input (syntax error)
+    }
+  }
+
+  render() {
+    // TODO: is this THE way to get this variable? it must be provided
```

Comment:
The architecturally correct way to get this variable is by getting it from the Redux store by setting it in the `mapStateToProps` function.

Reply:
This is how to do it: https://github.com/200ok-ch/organice/pull/154/commits/aa993c8827932544de12e580040e28e0257c0ff8